lib/req_ch.ex

```diff
@@ -3,117 +3,149 @@ defmodule ReqCH do
 
       It also accepts all formats described in the #{@formats_page} page.
 
       The `:explorer` format is special, and will build an Explorer dataframe
       in case the `:explorer` dependency is installed.
+
     * `:database` - Optional. The database to use in the queries.
       Default is `nil`.
 
   ## Examples
 
+  After setting a default database, one can make a request directly:
+
+      iex> req = ReqCH.new(database: "system")
+      iex> Req.post!(req, body: "SELECT number FROM numbers LIMIT 3").body
+      1\n2\n3\n
+
+  It's also possible to make a query using `Req.get/2`:
+
+      iex> req = ReqCH.new(database: "system")
+      iex> Req.get!(req, params: [query: "SELECT number FROM numbers LIMIT 3"]).body
+      1\n2\n3\n
+
+  In case the server needs authentication, it's possible to use `Req` options for that.
+
+      iex> req = ReqCH.new(base_url: "https://example.org:8123", auth: {:basic, "user:pass"})
+      iex> Req.post!(req, body: "SELECT number FROM system.numbers LIMIT 3").body
+      "0\\n1\\n2\\n"
+
+  """
+  @spec new(Keyword.t()) :: Req.Request.t()
+  def new(opts \\ []) do
+    attach(Req.new(base_url: "http://localhost:8123"), opts)
+  end
+
+  defp attach(%Req.Request{} = req, opts) do
+    req
+    |> Req.Request.prepend_request_steps(clickhouse_run: &run/1)
+    |> Req.Request.register_options(@options)
+    |> Req.Request.merge_options(opts)
+  end
+
+  defguardp is_query_params(value) when is_list(value) or is_map(value)
+
+  @doc """
+  Performs a query against ClickHouse API.
+
+  See docs from `new/1` for details about the options.
+
+  ## Examples
+
   Queries can be performed using both `Req.get/2` or `Req.post/2`, but GET
   is "read-only" and commands like `CREATE` or `INSERT` cannot be used with it.
+  For that reason, by default we perform a `POST` request.
+  To change that, use `query/2` with a pre-configured `req`.
 
   A plain query:
 
-      iex> req = Req.new() |> ReqCH.attach()
-      iex> Req.get!(req, clickhouse: "SELECT number FROM system.numbers LIMIT 3").body
+      iex> {:ok, response} = ReqCH.query("SELECT number FROM system.numbers LIMIT 3")
+      iex> response.body
       "0\\n1\\n2\\n"
 
   Changing the format to `:explorer` will return a dataframe:
 
-      iex> req = Req.new() |> ReqCH.attach()
-      iex> Req.get!(req, clickhouse: "SELECT number FROM system.numbers LIMIT 3", format: :explorer).body
+      iex> {:ok, response} = ReqCH.query("SELECT number FROM system.numbers LIMIT 3", [], format: :explorer)
+      iex> response.body
       #Explorer.DataFrame<
         Polars[3 x 1]
         number u64 [0, 1, 2]
       >
 
   Using parameters is also possible:
 
-      iex> req = Req.new() |> ReqCH.attach(format: :explorer, database: "system")
-      iex> Req.get!(req, clickhouse: {"SELECT number FROM numbers WHERE number > {num:UInt8} LIMIT 3", [num: 5]}).body
+      iex> opts = [format: :explorer, database: "system"]
```

Comment:
I think it'd be nice to show passing "regular" Req options that are important for CH like :base_url and :basic so it's clear that both ReqCH.new+Req.post and ReqCH.query are equivalent.

Reply:
I added one more example showing that. WDYT?

Reply:
Yup sounds good!